src/conn/conn_api.c

```diff
@@ -1084,6 +1084,44 @@ __conn_close(WT_CONNECTION *wt_conn, const char *config)
 			WT_TRET(wt_session->close(wt_session, config));
 		}
 
+	/*
+	 * Perform a system-wide checkpoint so that all tables are consistent
+	 * with each other.  All transactions are resolved but ignore
+	 * timestamps to make sure all data gets to disk.  Do this before
+	 * shutting down all the subsystems.  We have shut down all user
+	 * sessions, but send in true for waiting for internal races.
+	 */
+	if (!F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY) &&
+	    conn->ever_modified) {
```

Comment:
While I can reason about `ever_modified` and not wanting to mark the metadata dirty if nothing ever changed, if I remove that condition, I can successfully run `make check` and the entire Python test suite.  Is it just a minor optimization or is there some other failure you're thinking about here?

Reply:
I had the idea that we didn't write new files if the application never did any updates -- clearly I was mistaken, so I'll pull that code out.